src/main/java/org/folio/auth/authtokenmodule/storage/RefreshTokenStore.java

```diff
@@ -0,0 +1,199 @@
+package org.folio.auth.authtokenmodule.storage;
+
+import org.folio.auth.authtokenmodule.tokens.RefreshToken;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * The refresh token store provides persistence for refresh tokens. Refresh tokens need
+ * to be persisted so that: 1) we can ensure they are redeemed only once, and 2) we can
+ * take action (revoking all tokens for a user) when a given token is used more than once.
+ * When a refresh token is attempted to be redeemed more than once we consider it leaked.
+ */
+public class RefreshTokenStore extends TokenStore {
+  private static final Logger log = LogManager.getLogger(RefreshTokenStore.class);
+
+  private static final String REFRESH_TOKEN_SUFFIX = "refresh_tokens";
+
+  /**
+   * Constructs the refresh token store.
+   * @param vertx A reference to the current vertx object.
+   * @param tenant The tenant which is in scope for a given token. This can be obtained
+   * from the refresh token when the refresh token is created or arrives.
+   */
+  public RefreshTokenStore(Vertx vertx, String tenant) {
+    super(vertx, tenant);
+  }
+
+  /**
+   * Creates the table for this token store if it doesn't yet exist.
+   * @return A failed future should the save operation fail. Otherwise a succeeded future
+   * is returned, even if the table exists.
+   */
+  public Future<Void> createTableIfNotExists() {
+    log.info("Creating {} tables", RefreshTokenStore.class.getName());
+
+    // Refresh tokens have an owning user, but the token itself isn't persisted
+    // since it isn't used for anything. Just the id is enough.
+    String createTable = "CREATE TABLE IF NOT EXISTS " +
+        tableName(tenant, REFRESH_TOKEN_SUFFIX) +
+        "(id UUID PRIMARY key, user_id UUID NOT NULL, " +
+        "is_revoked BOOLEAN NOT NULL, expires_at INT8 NOT NULL)";
+
+    String createIndexExpiresAt = "CREATE INDEX IF NOT EXIST ON " +
```

Comment:
```suggestion
    String createIndexExpiresAt = "CREATE INDEX IF NOT EXISTS expires_at_idx ON " +
```
The same for the user_id index.

Reply:
Fixed